File: benchmarkDS/ourMethod/OPENAI_gpt-5-2025-08-07/Lse8.java
import java.util.regex.Pattern;

public class Lse8 {

    // Returns the value at the index specified by argv[1] within argv.
    // Throws IllegalArgumentException on invalid input.
    public static String valueAtIndex(String[] argv) {
        if (argv == null || argv.length < 2) {
            throw new IllegalArgumentException("insufficient arguments");
        }

        String idxStr = argv[1];
        // Only allow non-empty digits, cap length to avoid overflow
        if (idxStr == null || !Pattern.matches("\\d{1,10}", idxStr)) {
            throw new IllegalArgumentException("invalid index");
        }

        long idx;
        try {
            idx = Long.parseLong(idxStr);
        } catch (NumberFormatException ex) {
            throw new IllegalArgumentException("invalid index");
        }

        if (idx > Integer.MAX_VALUE) {
            throw new IllegalArgumentException("invalid index");
        }

        int i = (int) idx;
        if (i < 0 || i >= argv.length) {
            throw new IllegalArgumentException("index out of range");
        }

        return argv[i];
    }

    public static void main(String[] args) {
        String[][] tests = new String[][]{
            new String[]{"prog", "0"},
            new String[]{"app", "2", "x", "y", "z"},
            new String[]{"cmd", "4", "A", "B", "C", "D", "E"},
            new String[]{"tool", "-1", "val"},
            new String[]{"t", "10", "a", "b"}
        };

        for (int t = 0; t < tests.length; t++) {
            try {
                String result = valueAtIndex(tests[t]);
                System.out.println("Test " + (t + 1) + " -> " + result);
            } catch (IllegalArgumentException ex) {
                System.out.println("Test " + (t + 1) + " -> error: " + ex.getMessage());
            }
        }
    }
}

File: benchmarkDS/ZeroShot/OPENAI_gpt-5-2025-08-07/Lse8.cpp
#include <iostream>
#include <vector>
#include <string>

std::string getValueAtIndexFromArgs(const std::vector<std::string>& argv) {
    if (argv.size() < 2) {
        return std::string();
    }
    long long idx = 0;
    try {
        size_t pos = 0;
        idx = std::stoll(argv[1], &pos, 10);
        if (pos != argv[1].size()) {
            return std::string();
        }
    } catch (...) {
        return std::string();
    }
    if (idx < 0 || static_cast<size_t>(idx) >= argv.size()) {
        return std::string();
    }
    return argv[static_cast<size_t>(idx)];
}

int main() {
    // 5 test cases
    std::vector<std::vector<std::string>> tests = {
        {"prog", "2", "a", "b", "c"},   // expects "a"
        {"prog", "0"},                  // expects "prog"
        {"prog", "10", "x"},            // expects "" (out of range)
        {"prog", "-1", "x"},            // expects "" (negative index)
        {"prog", "notanint", "x"}       // expects "" (parse error)
    };

    for (const auto& t : tests) {
        std::string res = getValueAtIndexFromArgs(t);
        if (res.empty()) {
            std::cout << "" << std::endl;
        } else {
            std::cout << res << std::endl;
        }
    }
    return 0;
}